dev/plugins/hu.elte.txtuml.xtxtuml/src/hu/elte/txtuml/xtxtuml/formatting2/XtxtUMLFormatter.xtend

```diff
@@ -137,8 +138,14 @@ class XtxtUMLFormatter extends XbaseFormatter {
 		}
 	}
 
+	def dispatch void format(TUModifiers it, extension IFormattableDocument document) {
```

Comment:
Actually it can be done in a two-liner, but I can see your concerns. An example implementation:
```Java
def dispatch void format(TUModifiers mods, extension IFormattableDocument document) {
    #[TU_MODIFIERS__STATIC, TU_MODIFIERS__EXTERNALITY, TU_MODIFIERS__VISIBILITY]
        .map[mods.regionFor.feature(it)].sortBy[it?.offset].tail.forEach[prepend[oneSpace]];
}
```
Experimenting with the solution above I realized that with the current grammar in fact it's not possible to keep the aforementioned invariant, as for this we would need to put a space between the `modifiers` and `type` features of declaration prefixes. This can't be done because these are so-called _containment references_ and as such, they aren't supported by the region finder we use.

So the conclusion is that if we keep the initially proposed grammar, we have no other option than writing a comment about the violation and leaving the implementation unchanged. However, we still might have to consider the enum-based solution I mentioned in my [comment](https://github.com/ELTE-Soft/txtUML/pull/525#discussion_r125604117) about the grammar, which solution would also make my second proposal feasible here, as enum features are still supported by the region finder.